### PR TITLE
User registration

### PR DESCRIPTION
**Pull request by shinwachi**
### Please give a short summary of what's included in this submission
Added user registration controller using the email and password reset mechanism.  User can register via email, name, password, and user type.  You can't make admin user (defined in profile) using this mechanism.  

The newly minted user is "unverified" until the user gets the email with token and resets the password.  The registration and this password reset email verification mechanism is unit tested.

### Checklist
- [x] The Pull Request title describes the changes I've made
- [x] I've reviewed the **Files changed** tab
- [x] The code in the **Files changed** tab meets our style guides
- [x] New classes and public methods in the **Files changed** tab are documented
- [x] I've added two reviewers for this pull request

### CODE REVIEW

**Comment by stbenjam** [comment]
Relying on the password reset is clever, I think it works well.  Will the e-mail contain some link to activate the account? I guess the front end will need a page to do the activation.

File: src/main/java/edu/harvard/h2ms/web/controller/PasswordController.java [hunk 1]
```diff
@@ -105,11 +119,76 @@
...
+	 */
+	@RequestMapping(value = "/newuser/email", method = RequestMethod.POST)
+	public ResponseEntity<?> registerUserByEmail(@RequestBody Map<String, String> requestParams) {
+		
+		User user = new User();
+		user.setFirstName(requestParams.get("firstName"));
+		user.setMiddleName(requestParams.get("middleName"));
+		user.setLastName(requestParams.get("lastName"));
+		user.setEmail(requestParams.get("email"));
+		user.setPassword(requestParams.get("password"));
```
[the remark this answers is not in the document]
Comment (shinwachi) [reply src/main/java/edu/harvard/h2ms/web/controller/PasswordController.java]: Makes sense.  Changing.

**Comment by shinwachi** [comment]
Sorry, I made a typo somewhere along the upstream sync and messed up this PR. Canceling this one.

**Comment by shinwachi** [comment]
Ok, I *think* I have all the changes in.  Replaced the input with User bean (I hate the term) object as suggested by Stephen.  Dealt with password issue (password shouldn't be set by front end - not sure how to prevent this from happening from program, so front-end dev needs to be aware not to ask for password during registration).

**Comment by shinwachi** [comment]
thanks!

**Pull request merged**